src/app/searchsettings/searchsettings.component.spec.ts

```diff
@@ -1,14 +1,68 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
-
+import { Component, Input, Output } from '@angular/core';
+import { RouterTestingModule } from '@angular/router/testing';
+import { NavbarComponent } from '../navbar/navbar.component';
+import { CommonModule } from '@angular/common';
+import { FormsModule } from '@angular/forms';
+import { HttpModule, JsonpModule } from '@angular/http';
+import { StoreModule } from '@ngrx/store';
+import { StoreDevtoolsModule } from '@ngrx/store-devtools';
+import { AppComponent } from '../app.component';
+import { IndexComponent } from '../index/index.component';
+import { ResultsComponent } from '../results/results.component';
+import { NotFoundComponent } from '../not-found/not-found.component';
+import { AdvancedsearchComponent } from '../advancedsearch/advancedsearch.component';
+import { SearchBarComponent } from '../search-bar/search-bar.component';
+import { reducer } from '../reducers/index';
+import { FooterNavbarComponent } from '../footer-navbar/footer-navbar.component';
+import { AboutComponent } from '../about/about.component';
+import { ModalComponent } from 'ng2-bs3-modal/ng2-bs3-modal';
+import { ContactComponent } from '../contact/contact.component';
+import {InfoboxComponent} from "../infobox/infobox.component";
+import { RelatedSearchComponent } from '../related-search/related-search.component';
+import {KnowledgeapiService} from "../knowledgeapi.service";
```

Comment:
@harshit98 I imported all modules because otherwise, my tests kept being failed, and besides, in results and index, all these modules were imported. I also thought they will be useful for future use, but if you insist I can remove them.